services/hexpm/hexpm.service.js

```diff
@@ -96,52 +108,58 @@ class HexPmVersion extends BaseHexPmService {
   }
 }
 
-function DownloadsForInterval(downloadInterval) {
```

Comment:
There is a bit of refactoring going on here.

Previously, this service used the same pattern we see in https://github.com/badges/shields/pull/9732#discussion_r1390449157

Previously we defined a function that returns a class, and then called that function with `'day'`, `'week'` and `'all'` to generate 3 classes. One for each interval.

I don't think this really buys us much here. I converted this to a single class with an `:interval(dd|dw|dt)` param. This allows us to have a single example page and show interval as a drop-down menu, which is much more in line with how we do all our other downloads badges.

Reply:
Makes sense!